conjure-http/src/private/server.rs

```diff
@@ -35,6 +37,22 @@ where
     from_plain(&value, param)
 }
 
+pub fn path_param<T, D>(parts: &request::Parts, param: &str) -> Result<T, Error>
+where
+    D: DecodeParam<T>,
+{
+    let path_params = parts
+        .extensions
+        .get::<PathParams>()
+        .expect("PathParams missing from request");
+    let value = &path_params[param];
+    let params = value
+        .split('/')
+        .map(percent_encoding::percent_decode_str)
+        .map(|v| v.decode_utf8_lossy());
```

Comment:
Not 100% sure what we want the right behavior to be here. In principle, you *could* allow path parameters to be arbitrary byte sequences after percent-decoding, but as far as I can tell no mainstream URI builders are built to support that.

Query parameters are actually specced to do a lossy UTF8 conversion, but as far as I can tell the same is not true for path parameters. We could instead do a non-lossy decode and fail the conversion which may be preferable?